src/solver/conv_hip_implicit_gemm_fwd_xdlops.cpp

```diff
@@ -227,6 +233,49 @@ void RunCKSolution(const Handle& handle,
     }
 }
 
+template <typename DataType>
+InvokerFactory MakeInvokerFactoryHelper(CKArgs ck_args, size_t config_idx)
+{
+    return [ck_args = std::move(ck_args), config_idx](const std::vector<Kernel>& kernels) mutable {
+        std::ignore = kernels;
+
+        return [ck_args = std::move(ck_args),
+                config_idx](const Handle& handle, const AnyInvokeParams& primitive_parameters) {
+            RunCKSolution<DataType>(handle, primitive_parameters, ck_args, config_idx);
+        };
+    };
+}
+
+InvokerFactory MakeInvokerFactoryThrowError()
+{
+    return [](const std::vector<Kernel>& kernels) {
+        std::ignore = kernels;
+        MIOPEN_THROW(miopenStatusNotImplemented,
+                     "Convolution operation not implemented for this data type");
+        return [](const Handle&, const AnyInvokeParams&) {};
+    };
+}
+
+InvokerFactory
+MakeInvokerFactoryHipImplGemmFwdXdlops(const ProblemDescription& problem,
+                                       const PerformanceConfigHipImplicitGemmFwdXdlops& config)
+{
+
+    auto ck_args          = CKArgs{problem};
+    const auto config_idx = config.index;
+
+    switch(problem.conv_problem.GetInDataType())
+    {
+    case miopenInt8: return MakeInvokerFactoryHelper<int8_t>(std::move(ck_args), config_idx);
+    case miopenHalf: return MakeInvokerFactoryHelper<ck::half_t>(std::move(ck_args), config_idx);
+    case miopenFloat: return MakeInvokerFactoryHelper<float>(std::move(ck_args), config_idx);
+    case miopenInt32:
+    case miopenInt8x4:
+    case miopenBFloat16:
+    case miopenDouble: return MakeInvokerFactoryThrowError();
```

Comment:
```suggestion
    case miopenDouble: MIOPEN_THROW(miopenStatusInternalError, "ConvHipImplicitGemmFwdXdlops not implemented for this data type");
```

You should just fail rather than fail after compiling.
Plus this would mean an internal error in IsApplicable or usage inside library (even though it is recoverable) rather than some not supported parameters configuration provided by user.

Reply:
Implemented in 448628e